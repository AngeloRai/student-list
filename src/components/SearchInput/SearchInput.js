
const SearchInput = ({ setSearchWord, placeholder }) => {
  return (
    <input
      className="appearance-none focus:outline-none focus:border-b-1 focus:border-black border-b-2 p-2 w-full"
      placeholder={placeholder}
      onChange={(e) => setSearchWord(e.target.value)}
    />
  );
};

export default SearchInput;
