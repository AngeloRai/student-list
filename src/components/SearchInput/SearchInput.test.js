import React from "react";
import { shallow, mount } from "enzyme";
import SearchInput from "./SearchInput";
import toJson from "enzyme-to-json";

describe("tests search input component", () => {
  let wrapper;
  const setState = jest.fn();
  const mockPlaceholder = "John";
  const useStateSpy = jest.spyOn(React, "useState");
  useStateSpy.mockImplementation((init) => [init, setState]);

  beforeEach(() => {
    wrapper = mount(
      shallow(
        <SearchInput placeholder={mockPlaceholder} setSearchWord={setState} />
      ).get(0)
    );
  });

  it("renders without crashing", () => {
    shallow(<SearchInput />);
  });

  it("checks for changes in the code", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("accepts props", () => {
    const mockPlaceholder = "John";
    const mockHandleChange = jest.fn();
    const wrapper = mount(
      <SearchInput
        placeholder={mockPlaceholder}
        handleChange={mockHandleChange}
      />
    );
    expect(wrapper.props().placeholder).toBe(mockPlaceholder);
    expect(wrapper.props().handleChange).toBe(mockHandleChange);
  });

  it("Should capture search word correctly correctly onChange", () => {
    const searchWord = wrapper.find("input").at(0);
    searchWord.instance().value = "Test";
    searchWord.simulate("change");
    expect(setState).toHaveBeenCalledWith("Test");
  });
});
