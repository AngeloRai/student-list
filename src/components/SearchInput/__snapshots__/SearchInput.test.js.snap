// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`tests search input component renders without crashing 1`] = `
<input
  className="appearance-none focus:outline-none focus:border-b-1 focus:border-black border-b-2 p-2 w-full"
  onChange={[Function]}
  placeholder="John"
/>
`;
